### PR TITLE
Change inter store requisition interface design

### PR DESCRIPTION
**Pull request by edmofro**
Major changes:
- Separated customer and supplier requisitions into pages with buttons
in appropriate sections of the menu
- Customer invoice no longer created automatically on syncing in a
response requisition
- Customer invoice created (and finalised) on finalising response
requisition

And fix a few things along the way:
- Add customer invoice number to customer requisition info section
- Change ‘Supplying Store’ to customer in customer requisition info
section
- Fix date formatting in ‘Entered Date’ everywhere
- Remove unhelpful ‘Entered By’ on both type of requisition
- Changed finalise message to be different for customer and supplier
requisitions

### CODE REVIEW

**Comment by edmofro** [comment]
I still need to:
- [ ] Add a sensible enteredBy
- [ ] Work out why names are unreliably syncing
- [ ] Test further downstream: do the customer invoices generated from requisitions create a supplier invoice in the other store?

**Comment by nmadruga** [comment]
Wow, that's looking way more user friendly. The Customer Invoice be created just after Customer Requisition is finalised very good. The Labels now make sense: fix #472 and fix #471. I don't know if I like the formatting on the dates yet. I didn't see the difference and it's cutting the time always. Also the customer name is cut, maybe we cold give more space there?
screenshot_20170818-104906
screenshot_20170817-151955
But besides those minor issues for me that change is wonderful. It solves the problem fix #473

**Comment by edmofro** [comment]
Thanks @nmadruga! I agree about the customer name, and will work on that. However, for the entry date the fix was actually in another library (see [sussol-utilities](https://github.com/sussol/sussol-utilities/commit/a0b5151ccf7a7ffd1a36c150a5e8029316745b38)), so you just need to run `yarn` and it will be fixed :-)

**Comment by Chris-Petty** [comment]
@Nicole just FYI fix #someIssueNumber doesn't work unless it is in the top comment or in the comment/commit description box when trying to merge.

File: src/localization/modalStrings.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ export const modalStrings = new LocalizedStrings({
     edit_the_requisition_comment: 'Edit the requisition comment',
     edit_their_reference: 'Edit their reference',
     finalise_customer_invoice: 'Finalise will lock this invoice permanently.',
-    finalise_requisition: 'Finalise will send this requisition and lock it permanently.',
+    finalise_supplier_requisition: 'Finalise will send this requisition and lock it permanently.',
+    finalise_customer_requisition: 'Finalise will generate a finalised customer invoice, adjust inventory, and lock this requisition permanently.',
```
Comment (Chris-Petty) [inline src/localization/modalStrings.js]: Is this up to date with this implementation? The customer invoice already exists yes? To the user it doesn't exist sure.

Speaking of which, is it possible to just hold off assigning a serial number until the related requisition is finalised?

File: src/database/UIDatabase.js [hunk 2]
```diff
@@ -7,18 +7,27 @@ export class UIDatabase {
...
-        return results.filtered(
-          "isVisible == true AND isSupplier == true AND type == 'facility'"
-        );
+        return results.filtered('isVisible == true AND isSupplier == true AND type == "facility"');
       case 'Item':
         return results.filtered('isVisible == true');
+      case 'RequestRequisition':
+        return results.filtered('type == "request"');
+      case 'ResponseRequisition':
+        return results.filtered('serialNumber != "-1" AND type == "response"');
```
Comment (Chris-Petty) [inline src/database/UIDatabase.js]: Breaks post sync processor again I believe. This is the right way to do this. I'll commit my suggested fix in a moment.

Comment (edmofro) [inline src/database/UIDatabase.js]: Doesn't break post sync, uses different term. Is the right way to do it.

File: src/localization/modalStrings.js (hunk 1 above)
Comment (edmofro) [inline src/localization/modalStrings.js]: Yes up to date with implementation. Doesn't exist to user, wording is user facing. When training, I think you should never mention that it exists under the hood...that's complicated. To them, we create it when they finalise the requisition.

Comment (edmofro) [inline src/localization/modalStrings.js]: Not sure what your serial number point is, but you can try anything you like in the weekend ;-)

**Pull request merged**